website/static/js/logFeed.js

```diff
@@ -1,238 +1,245 @@
+                    // function defined inside loop
+                }
+                ctrl.paginators().push({
+                    text: '...',
+                    url: function() { }
+                });
+            }
+            // one ellipse at the beginning
+            else if (ctrl.currentPage() > ctrl.totalPages - MAX_PAGES_ON_PAGINATOR_SIDE) {
+                ctrl.paginators().push({
+                    text: '...',
+                    url: function() { }
+                });
+                for (i = ctrl.totalPages - MAX_PAGES_ON_PAGINATOR_SIDE; i < ctrl.totalPages - 1; i++) {
+                    ctrl.paginators().push({
+                        text: i + 1,
+                        url: function() {
+                            ctrl.pageToGet(parseInt(this.text));
+                            if (ctrl.pageToGet() !== ctrl.currentPage()) {
+                                return _buildLogUrl(ctrl.node, ctrl.pageToGet());
+                            }
+                        }
+                    });/* jshint ignore:line */
+                    // function defined inside loop
+                }
+            }
+            // two ellipses
+            else {
+                ctrl.paginators().push({
+                    text: '...',
+                    url: function() { }
+                });
+                for (i = parseInt(ctrl.currentPage()) - 1; i <= parseInt(ctrl.currentPage()) + 1; i++) {
+                    ctrl.paginators().push({
+                        text: i + 1,
+                        url: function() {
+                            ctrl.pageToGet(parseInt(this.text));
+                            if (ctrl.pageToGet() !== ctrl.currentPage()) {
+                                return _buildLogUrl(ctrl.node, ctrl.pageToGet());
+                            }
+                        }
+                    });/* jshint ignore:line */
+                    // function defined inside loop
+                }
+                ctrl.paginators().push({
+                    text: '...',
+                    url: function() { }
+                });
+            }
+            // last page
+            ctrl.paginators().push({
+                text: ctrl.totalPages,
+                url: function() {
+                    ctrl.pageToGet(ctrl.totalPages);
+                    if (ctrl.pageToGet() !== ctrl.currentPage()) {
+                        return _buildLogUrl(ctrl.node, ctrl.pageToGet());
+                    }
+                }
+            });
+            // next page
+            ctrl.paginators().push({
+                url: function() { return ctrl.nextPage; },
+                text: '>'
+            });
+        }
 
-////////////////
-// Public API //
-////////////////
+        return m('.db-activity-list.m-t-md', [
 
-var defaults = {
-    /** Selector for the progress bar. */
-    progBar: '#logProgressBar'
-};
+            ctrl.failed ? m('p', 'Unable to retrieve logs at this time.') :
```

Comment:
For consistency with other error messages on the site, it would be good to include a link to mail OSF support.